guides/release/templates/handlebars-basics.md

```diff
@@ -1,85 +1,203 @@
 
 ```javascript {data-filename=app/controllers/application.js}
 import Controller from '@ember/controller';
 
 export default Controller.extend({
   firstName: 'Trek',
-  lastName: 'Glowacki'
+  lastName: 'Glowacki',
+  favoriteFramework: 'Ember'
 });
 ```
 
+The attributes like `firstName` can be used in the template
+by putting them inside of curly braces, plus the word
+`this`:
+
+```handlebars {data-filename=app/templates/application.hbs}
+Hello, <strong>{{this.firstName}} {{this.lastName}}</strong>!
+```
+
 The above template and controller render as the following HTML:
 
 ```html
 Hello, <strong>Trek Glowacki</strong>!
 ```
 
-Remember that `{{this.firstName}}` and `{{this.lastName}}` are bound data. That means
-if the value of one of those properties changes, the DOM will be updated
-automatically.
+If you use JavaScript to change the values of `firstName` or `lastName` in the Controller, what the user sees will be automatically updated! In Ember, this is referred to as "data binding."
 
-As an application grows in size, it will have many templates backed by
-controllers and components.
+Here's another example using a Component.
+Components are usually made of two files, a JavaScript file `my-component.js` and a
+template file with the same name, `my-component.js`. Whatever attributes are defined
+in the Component's JavaScript are available for use in the Template:
 
-### Helpers
+```javascript {data-filename=app/components/my-component.js}
+import Component from '@ember/component';
 
-Ember Helpers are functions that can compute values and can be used in any template.
+export default Component.extend({
+  firstName: 'Jessica',
+  lastName: 'Jordan'
+});
+```
 
-Ember gives you the ability to [write your own helpers](../writing-helpers/), to bring a minimum of logic into Ember templating.
+```handlebars {data-filename=app/templates/components/my-component.hbs}
+Hello, <strong>{{this.firstName}} {{this.lastName}}</strong>!
+```
+
+This would render:
+
+```html
+Hello, <strong>Jessica Jordan</strong>!
+```
+
+## Things you might see in a Template
+
+A lot more than just HTML markup can go in Templates. In the other pages of this guide, we will cover the features one at a time. In general, special Ember functionality will appear inside curly braces, like this: `{{example}}`. Here are a few examples of Ember Handlebars in action:
+
+Route example:
+```hbs {data-filename=app/templates/application.hbs data-update=true}
+{{!-- Example of a comment that will be invisible --}}
+
+{{!-- outlet determines where a child route's content should render. Don't delete it until you know more about it! --}}
+<div>
+  {{outlet}}
+</div>
+
+{{!-- One way to use a component within a template. This is called "Angle Bracket Component syntax." --}}
+<MyComponent />
+
+```
+
+Component example:
+```hbs {data-filename=app/components/templates/my-component.hbs data-update=true}
+{{!-- An attribute that is defined in a component's JavaScript file --}}
```

Comment:
```suggestion
{{!-- A property that is defined in a component's JavaScript file --}}
```

Reply:
Based on @locks suggestion above. I agree, we should try to use "attribute" consistently to mean only HTML attributes

Reply:
done